### PR TITLE
VxAdmin: Improve configuration error handling

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview
Fixes: #4290 

Previously, the screen would just hang if there was an error parsing the zip file. Now, we catch zip file errors and show a message.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/530106/d8f53e0e-ce54-494e-899b-279446136ed9

## Testing Plan
Manual test, automated tests

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/frontend/src/utils/initial_setup_package.ts [hunk 1]
```diff
@@ -21,29 +23,37 @@ export interface InitialAdminSetupPackage {
...
+  }
 
   // TODO(kofi): Import translation/audio files as well.
 
-  return {
+  return ok({
     electionString,
     systemSettingsString,
-  };
+  });
```
Comment (eventualbuddha) [inline apps/admin/frontend/src/utils/initial_setup_package.ts:52]:
```suggestion
): Promise<Result<InitialAdminSetupPackage, ConfigureError>> {
  try {
    const zipfile = await openZip(source);
    const entries = getEntries(zipfile);
    const electionEntry = getFileByName(
      entries,
      BallotPackageFileName.ELECTION
    );
    const electionString = await readTextEntry(electionEntry);
    const systemSettingsEntry = getFileByName(
      entries,
      BallotPackageFileName.SYSTEM_SETTINGS
    );
    const systemSettingsString = await readTextEntry(systemSettingsEntry);

    // TODO(kofi): Import translation/audio files as well.

    return ok({
      electionString,
      systemSettingsString,
    });
  } catch (error) {
    return err({ type: 'invalidZip', message: String(error) });
  }
```

Comment (jonahkagan) [reply apps/admin/frontend/src/utils/initial_setup_package.ts]: Thanks 30f08dde4

File: apps/admin/frontend/src/screens/unconfigured_screen.tsx [hunk 2]
```diff
@@ -81,12 +78,17 @@ export function UnconfiguredScreen(): JSX.Element {
...
-        await readInitialAdminSetupPackageFromFile(file);
-      await configureFromElectionPackage(
-        initialSetupPackage.electionString,
-        initialSetupPackage.systemSettingsString
-      );
+      const packageResult = await readInitialAdminSetupPackageFromFile(file);
+      if (packageResult.isErr()) {
+        setConfigureError(packageResult.err());
+        // eslint-disable-next-line no-console
+        console.error(packageResult.err().message);
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/unconfigured_screen.tsx:85]: For consistency. But my guess is that the above log is for debugging, so maybe this one is too

File: apps/admin/frontend/src/screens/unconfigured_screen.tsx [hunk 3]
```diff
@@ -107,6 +109,8 @@ export function UnconfiguredScreen(): JSX.Element {
           <Icons.Danger color="danger" />{' '}
           {(() => {
             switch (configureError.type) {
+              case 'invalidZip':
+                return 'Invalid election package zip file.';
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/unconfigured_screen.tsx:113]: Yeah, I'm going to be cleaning that up today by making it say "election package" everywhere, so I was just getting a head start.

**Pull request merged**